CA_DataUploaderLib/ServerUploader.cs

```diff
@@ -94,19 +109,30 @@ private void SendVector(DataVector vector)
                 SendEvent(this, new EventFiredArgs(e.Data, e.EventType, e.TimeSpan.AddTicks(eventIndex++)));
 
             //now queue vectors
-            _vectorsChannel.Writer.TryWrite(WithOnlyUploadFields(vector));
+            _vectorsChannel.Writer.TryWrite(FilterOnlyUploadFieldsAndCheckInvalidValues(vector, out var invalidValueMessage));
+
+            //Possibly send event due to invalid values detected in the vector
+            if (!string.IsNullOrWhiteSpace(invalidValueMessage))
+                SendLowFrequencyEvent(new EventFiredArgs("Warning - invalid values detected: " + invalidValueMessage[2..], EventType.Log, DateTime.UtcNow));
 
-            DataVector WithOnlyUploadFields(DataVector fullVector)
+            DataVector FilterOnlyUploadFieldsAndCheckInvalidValues(DataVector fullVector, out string invalidValueMessage)
             {
                 var fullVectorData = fullVector.Data;
                 var uploadData = new double[uploadDesc.Length];
                 int j = 0;
+                StringBuilder invalidValues = new();

```

Comment:
This allocates memory on every vector even when not tripping

Reply:
fixed